chroot-bin/croutoncycle

```diff
@@ -307,6 +328,15 @@ else
         # display will be stuck on the old vt.
         sudo -n chvt 2
         sudo -n chvt "$dest"
+    else
+        dest="/tmp/.X${destdisp#:}-lock"
+        if [ -f "$dest" ]; then
+            # Trigger the target before releasing the current owner
+            kill -USR1 `cat /tmp/.X${destdisp#:}-lock`
+        fi
+        if [ "${freonowner:-0}" != 0 ]; then
+            kill -USR1 "$freonowner"
+        fi
```

Comment:
If I remember correctly how switching works (X server crashes if it cannot obtain DRM master), this feels race-prone (I'm actually surprised this works at all...)... Does it work if you add a sleep in between the 2 statements?